src/config.rs

```diff
@@ -1,18 +1,114 @@
 use bitcoin::network::constants::Network;
-use clap::{App, Arg};
 use dirs::home_dir;
 use num_cpus;
 use std::fs;
+use std::fmt;
 use std::net::SocketAddr;
 use std::net::ToSocketAddrs;
-use std::path::{Path, PathBuf};
+use std::path::PathBuf;
 use std::sync::Arc;
+use std::str::FromStr;
+use std::ffi::{OsStr, OsString};
+use std::convert::TryInto;
 use stderrlog;
 
 use crate::daemon::CookieGetter;
 use crate::errors::*;
 
-const DEFAULT_SERVER_ADDRESS: &str = "127.0.0.1"; // by default, serve on IPv4 localhost
+const DEFAULT_SERVER_ADDRESS: [u8; 4] = [127, 0, 0, 1]; // by default, serve on IPv4 localhost
+
+mod internal {
+    #![allow(unused)]
+
+    include!(concat!(env!("OUT_DIR"), "/configure_me_config.rs"));
+}
+
+pub enum AddressError {
+    InvalidUtf8(OsString),
+    ResolvError { addr: String, err: std::io::Error },
+    NoAddrError(String),
+}
+
+impl fmt::Display for AddressError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            AddressError::InvalidUtf8(val) => write!(f, "{:?} isn't a valid UTF-8 sequence", val),
+            AddressError::ResolvError { addr, err } => write!(f, "Failed to resolve address {}: {}", addr, err),
+            AddressError::NoAddrError(addr) => write!(f, "No address found for {}", addr),
+        }
+    }
+}
+
+
+#[derive(Deserialize)]
```

Comment:
Oh, this doesn't resolve the address. I think it is a bad idea to try to resolve the address when parsing/deserializing - leads to confusing code. We still want a newtype to have a better description, but we should resolve instead of converting.